nodebuilder/state/service.go

```diff
@@ -17,6 +17,8 @@ import (
 type Module interface {
 	// IsStopped checks if the Module's context has been stopped
 	IsStopped() bool
+	// ReadOnly checks if the Module has disabled write access
+	ReadOnly() bool
```

Comment:
Our module interfaces are soon to be used by the RPC. This means that all the methods are exposed over RPC. This type of _hack_ is not aligned with that. Also, note that this `ReadOnly` is only used by the middleware, but not in the case we use the module interface directly as a lib, which we defined as the main use case. 

I recommend closing this PR and revisiting this issue later once the RPC work is done.

Also note that current "RPC/API" endpoints will become a gateway soon, which means it will be read-only by default, and sending tx transactions from the node's key through the key must be forbidden.